src/azure-cli/azure/cli/__main__.py

```diff
@@ -40,6 +40,9 @@ def cli_main(cli, args):
 
     exit_code = cli_main(az_cli, sys.argv[1:])
 
+    if az_cli.result.error is not None and not telemetry.has_exceptions():
```

Comment:
when has_exceptions(), the exceptions will be logged also, but not to telemetry exception right?

Reply:
when has_exceptions(), it means some raised exception is manually added to telemetry session already, so here no need to add it again.